net/modemmanager/files/usr/libexec/modemmanager/ModemManager-watcher

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
+. /usr/share/ModemManager/modemmanager.common
+
+MODEM_PATH="$1"
+
+watcher_clean_up() {
+	# This step is necessary to terminate all pending child pipelines
+	# (especially mmcli --monitor-state)
+	child_pids=$(pgrep -P $$)
+	if [ -n "${child_pids}" ]; then
+		mm_log "debug" "Cleanup watcher pids: $child_pids"
+		kill -9 "$child_pids" 2>/dev/null
+	fi
+}
+
+main() {
+	local disconnect_detected=0
+
+	mm_log "info" "Start modem state watcher for $MODEM_PATH"
+
+	trap watcher_clean_up INT TERM
+
+	while read -r line; do
+		mm_log "debug" "State changed: $line"
+		if echo "$line" | grep -q "'connected' --> 'disconnecting'"; then
```

Comment:
I had some disconnects with another state transition: `state changed (connected -> registered)`. They have to be handled too.

This state transition has been caused by the following reasons:

- mmgdsi-card-event
- implicitly-detached

Reply:
Every transition out of "connected" is a disconnection